Update os_server.py to include option for files

userlerueda opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Fixes #22244 by adding files options to os_server.py module to allow files to be passed to instance when config-drive option is present.

Supersedes #22449

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
os_server.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.2.0.0
  config file = /vms-3.0/ansible/ansible.cfg
  configured module search path = ['./modules/']
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

userlerueda committed:
Update os_server.py to include option for files

Adds files options to os_server.py module to allow files to be passed to instance when config-drive option is present.

JonathanO committed:
Restore SIGPIPE handler to DFL on POpen

Python sets the SIGPIPE handler to SIG_IGN. On execv() signal handlers are
reset to their defaults, EXCEPT those that are SIG_IGN which are left ignored.
In Python 3 subprocess.popen explicitly resets the SIGPIPE handler to SIG_DFL,
but unfortunately in Python 2.7 it does not. This leads to subprocesses being
executed with SIGPIPE ignored. This is often a problem with bash scripts which
rely on SIGPIPE to terminate commands in a pipe, but can easily be a problem
with other applications.

This implements the Python 3 behaviour for Python 2.7 by using a preexec_fn.

sirredbeard committed:
Update windows_faq.rst

<!--- Your description here -->

+label: docsite_pr

Windows Subsystem for Linux is now officially supported by Microsoft. It is out of beta and will be shipping on Windows Sever 2019.

jborean93 committed:
fixed tests after EAP set to Stop globally, ci_complete

ckotte committed:
VMware: Add check mode support in vmware_host_config_facts (#46272)

yannrouillard committed:
Only delete key from redis in-memory cache if present (#35126)

Fixes #35120 : the redis cache plugin keeps key/value
entries in an in-memory cache to avoid hitting the
redis database each time.

The problem is that a cache entry is only set when
a value is get or set but it is always deleted when
trying to delete a value.

When the --flush-cache ansible-playbook option is used,
the redis cache plugin is first asked to remove every
entry corresponding to every hostname present in the inventory.
As no value as been set/get so far, it then tries to delete
an unexisting value from the cache and hence crashes with
a KeyError exception.

haidaraM committed:
Docsite: Add a new tool:  Ansible Playbook Grapher (#46435)

Add Ansible Playbook Grapher: a tool create a graph representing your Ansible playbook tasks and roles.

https://github.com/haidaraM/ansible-playbook-grapher

+label: docsite_pr

major committed:
Docs: Cleanup selinux_permissive module docs (#46351)

Update the selinux_permissive instructions and add a comment to the
provided example.

Signed-off-by: Major Hayden <major@redhat.com>

mattclay committed:
Use default-test-container version 1.3.0.

newtonne committed:
Macports: Add upgrade parameter and replace update_ports with selfupdate (#45049)

* macports: Replace update_ports with selfupdate

- Macports discourages use of `port sync` and recommends using `port
selfupdate` instead.
- Keep `update_cache` and `update_ports` as aliases.
- No longer require the `name` parameter so that `selfupdate` can be
used in a task by itself.

* macports: Add upgrade parameter

- New upgrade parameter which can be used to upgrade all outdated ports.

* Add changelog fragment

beeankha committed:
Update asa_config.py (#46448)

<!--- Your description here -->

+label: docsite_pr

dagwieers committed:
Docs: Clean up of 'file' module docs (#46327)

This is part of a series of module doc cleanups.

major committed:
Docs: iptables module cleanup (#46368)

Signed-off-by: Major Hayden <major@redhat.com>

felixfontein committed:
Subscribing to notifications. (#46372)

jborean93 committed:
win_chocolatey: remove test packages after tests are run (#46431)

jborean93 committed:
ps-lint: ignore rules that are not relevant to Ansible (#46376)

dagwieers committed:
Docs: Clean up of 'assemble' module docs (#46328)

This is part of a series of module doc cleanups.

major committed:
Docs: Cleanup make module docs (#46422)

Signed-off-by: Major Hayden <major@redhat.com>

samdoran committed:
Use proper index value with insertbefore on a one line file (#46071)

Add tests and changelog

mattclay committed:
Add file exists check in integration-aliases test.

sugitk committed:
remove unnecessary space (#46462)

kindly advised from jborean93

dagwieers committed:
Docs: Clean up 'win_service' module docs (#46407)

* Docs: Clean up 'win_service' module docs

This is part of a series of module doc cleanups.

* Fix typo pause vs paused

* Use both YAML list syntaxes

Im0 committed:
Lookup plugin for rabbitmq (#44070)

* Adding a basic get lookup for rabbitmq.

* Always return a list

* If content type is JSON, make accessible via dict.

* Fixed incorrect json.loads variable and missing raise

* Change to document returned data

* Fixed pep8 issues

* Adding integration testing

* Moving lookup intgration tests to new target

* New rabbitmq lookup plugin (#44070).

* New rabbitmq lookup plugin (#44070).

* PR review feedback updates

* Testing pika is installed

* Minor mods to tests

* Check if connection is already closed or closing

* Updated tests and connection testing

* PR review feedback updates

* PR review include ValueError in AnsibleError output

* Suggesting to use set_fact when using returned variable more than once.

* Cleaned up some tests, added some notes and handling connection closure on some exceptions.

* Removed finally statement and added some additional error handling.

* Added some additional error handling.

* PR review updates.

* Additional integration tests and removing return in finally

* Updated version

* Changing back to running tests on ubuntu.

* Additional tests

* Running tests on  Ubuntu only

* Fixing syntax error

* Fixing ingtegration tests and a string/byte issue

* Removed non-required test and fixed BOTMETA

* Trying to fix integration test failure on ubuntu1404

* Some issues occured when handling messages from the queue with to_native.  Switching to to_text resolved the issues.

* Renaming channel to queue (thanks dch). Disabling trusty tests.

adrw committed:
Fix spelling error in subelements Filter example (#46446)

<!--- Your description here -->

+label: docsite_pr

acozine committed:
updates latest in docs conf.py (#46400)

dagwieers committed:
Docs: Clean up of 'template' module docs (#46297)

* Docs: Clean up of 'template' module docs

* Changed influenced by review comments

samccann committed:
add simple j2 example to vyos_config (#46442)

dagwieers committed:
Docs: Clean up 'win_reboot' module docs (#46377)

* Docs: Clean up 'win_reboot' module docs

This is part of a series of module doc cleanups.

* Remove hypothetical example

dagwieers committed:
Docs: Clean up of 'fetch' module docs (#46330)

* Docs: Clean up of 'fetch' module docs

This is part of a series of module doc cleanups.

* Fixes as suggested by review

ganeshrn committed:
Make default_operation optional in netconf_config module (#46333)

Fixes #46257

*  As per netconf rfc default-operation value is optional,
   hence removing the default value for default_operation
   option.

mattclay committed:
Improve ansible-test environment checking between tests. (#46459)

* Add unified diff output to environment validation.

This makes it easier to see where the environment changed.

* Compare Python interpreters by version to pip shebangs.

This helps expose cases where pip executables use a different
Python interpreter than is expected.

* Query `pip.__version__` instead of using `pip --version`.

This is a much faster way to query the pip version. It also more
closely matches how we invoke pip within ansible-test.

* Remove redundant environment scan between tests.

This reuses the environment scan from the end of the previous test
as the basis for comparison during the next test.

ckotte committed:
VMware: Add check mode support to module vmware_host_acceptance (#46260)

* Improve module description
* Add check mode support

ckotte committed:
VMware: Add check mode support to module vmware_host_firewall_facts (#46265)

mattclay committed:
Add symlinks sanity test. (#46467)

* Add symlinks sanity test.
* Replace legacy test symlinks with actual content.
* Remove dir symlink from template_jinja2_latest.
* Update import test to use generated library dir.
* Fix copy test symlink setup.

tstoner committed:
Purging logging configuration before running playbook (#46302)

dagwieers committed:
Fix subelements example correctly (#46482)

acozine committed:
updates language about documentation versions and releases on index page (#46465)

samdoran committed:
Add documentation on underlying tools in user module (#46455)

jborean93 committed:
added porting guide for PowerShell EAP changes (#46451)

dagwieers committed:
win_reboot: Fix for ignore post_reboot_delay (#46360)

* win_reboot: Fix for ignore post_reboot_delay

This fixes an issue where win_reboot would be ignoring the provided
post_reboot_delay (and on Windows timing/waiting is everything!)

This must be backported to the v2.7 branch.

* Merge post-reboot handling into run()

briantist committed:
Fix support for SubnetMappings and EIPs in NLB (#42979)

* Fix support for SubnetMappings and EIPs in NLB

* Fix style failures

blamarvt committed:
[aws] ALB/NLB support SubnetMappings in AWS create LB call (#42060)

willthames committed:
Add test suite for ec2_vpc_igw before boto3 upgrade (#45903)

mnecas committed:
add external provider to ovirt network (#43636)

* add external provider to ovirt network

* add version added

* add example

* test of importing external network

* ovirt correct datacenter search in external provider

* repair importing external provider

* update external provider docs

* update pep8 syntax

mnecas committed:
Ovirt add kernel persist (#45555)

* ovirt add kernel persist

* Add description to kernel_params_persist

* update kernel_params_persist control logic

* correct pep8 syntax

* add condition for kernel_params

* put kernel persist check to function

* kernel persist check update tabs

jorijinnall committed:
fix drain mode 'bool' (#30833) (#46478)

jborean93 committed:
shippable: fix py3 issues with downloader (#46522)

mattclay committed:
Python 3.8 collections compatibility fixes.

Includes a new pylint blacklist plugin to prevent regressions.

mnecas committed:
Ovirt delete snapshots after x days  (#45551)

* ovirt add snapshot delete after x days

* update logic of deleting snapshot and rename it

* update docs and names of vars

* update docs syntax

* Update ovirt_snapshot.py

* put remove logic to function

* use default removing function for snapshot

jborean93 committed:
azure_rm_common: use __version__ from module_util (#46184)

rhodain committed:
Ovirt_disk - Add ability to refresh passthrough disks (#46386)

* Ovirt_disk - Add ability to refresh passthrough disks

A new paramter "host" is added. When this paramter is specified the newly
created disk or an existing disk will refresh its information about the
underlying storage via the specified host. This oprion is only valid for
passthrough disks. This option requires at least the logical_unit.id to be
specified.

* Host search modification and some other minor changes.

* pep8 correction

dagwieers committed:
mail: Fix regression when sending mail without SSL (#46403)

When refactoring the code the non-SSL use-case was broken.

The main cause is that we have no way to do integration tests for
testing SMTP.

xuyixin1996 committed:
zabbix_host: fix link template error (#46521)

* add host interface before link template

mattclay committed:
Move docsite requirements to the docs/docsite dir. (#46518)

Akasurde committed:
AWS: check dbname attribute before accessing (#46253)

Check dbname attribute in instance before accessing it in RDS module

Fixes: #38210

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Fix parametrize warning in unit tests.

Akasurde committed:
Add missing bracket in documentation (#46564)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
docker_container: fix behavior when image is not specified (#46322)

* Don't simply ignore container in present() if image is not specified.

* Use image from existing container for recreation if not specified.

* Added changelog.

* Improve comment.

felixfontein committed:
docker_container, docker_image_facts: allow to use image IDs (#46324)

* Allow to specify images by hash for docker_container and docker_image_facts.

* flake8

* More sanity checks.

* Added changelog.

* Added test.

* Make compatible with Python < 3.4.

* Remove out-commented imports.

remyleone committed:
Add region to the volume facts (#46214)

remyleone committed:
Add tags grouping (#45741)

ckotte committed:
VMware: Add check mode support to module vmware_host_dns_facts (#46530)

Akasurde committed:
Misc typo in logentries plugin module (#46568)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

VitSkakun committed:
fix digitalocean inventory tags endpoint (#46192)

remyleone committed:
Add region to snapshots facts (#46213)

remyleone committed:
Add region to server facts (#46212)

remyleone committed:
Add region parameters to sg (#46211)

remyleone committed:
Specify region in IP facts (#46210)

BondAnthony committed:
Digital Ocean firewall facts (#46582)

evertmulder committed:
Added module to manage manageiq tenants (#43988)

* Added module to manage manageiq tenants

* Fixed pep8 and module validation issues for manageiq_tenant

* Changes based on review
- Added parent attribute to specify the parent
- Cleanup return of tenant
- Update documentation on quotas

* Updated version added to 2.8

flowerysong committed:
Port the sns module to boto3 (#45634)

* Port sns to boto3

* Exception handling for ARN lookup

* sns: Add integration tests

Medzoner committed:
proxmox: doc: Fix example for restart container (#46602)



+label: docsite_pr

jborean93 committed:
Win chocolatey facts module (#46610)

* add module win_chocolatey_facts

* rename example name

* fix ansible-test errors

* add integration tests

* fix integration test

* implementation of improvement proposals

* implementation feedback

* implementation feedback

* fix trailing-whitespace

* implementation feedback

* fix version

* fix lint

* add test targets

* Updated modules docs and tests

Co-authored-by: Simon Baerlocher <sbaerlocher@users.noreply.github.com>

silid committed:
Add chdir support to win_package (#46123)

* Add chdir support to win_package

* Add chdir support to uninstall too.

* Update docs to account for uninstall.

* Correct variable substitution for hashtable.

* added changelog fragment

trishnaguha committed:
nxos_evpn_vni check_mode (#46612)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

felixfontein committed:
docker_container: improve publish all ports functionality (#46594)

* Improve handling of published_ports: all.

* Add changelog.

felixfontein committed:
docker_container: fix problem with published_ports idempotency (#46595)

* Fix comparisons for expected_ports (set vs dict).

* Added changelog.

Akasurde committed:
VMware: new module vmware_host_feature_facts (#45966)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
docker_container: allow more mount modes for volumes (#46598)

* Being more strict about volume mount modes.

ckotte committed:
VMware: Add check mode support to module vmware_host_service_manager (#46271)

ckotte committed:
VMware: Add check mode support to module vmware_host_ntp (#46268)

felixfontein committed:
docker_container: add port range and IPv6 support for published_ports (#46596)

* Allow port ranges.

* Adding IPv6 support for published_ports.

* Die when hostname is passed instead of IP address.

* Added changelog.

gundalow committed:
changelogs/fragments/: community (#46616)

Commit:
Account for boolean OVF properties (#45529)

Currently, Ansible interprets variables with a True|False value as
boolean. This causes the vmware_deploy_ovf module to break, because
it can only accept string values as properties.

This fix checks if a value is boolean, and converts it to a string
if it is. Since integers do not seem to be causing the same error, this
is the only check we appear to need.

After completion, OVF properties that are boolean can be specified as
yes|no or true|false.

Closes: #45528

agx committed:
influxdb_retention_policy: don't fail with empty error message (#28792)

On connection errors the influxdb python module raises an exception with
an empty message leading to a fail_json with an empty error message:

    https://github.com/influxdata/influxdb-python/blob/265d14736bb739aed0f357d8da248d51f549244b/influxdb/influxdb08/client.py#L252

Fail with a more descriptive error in this case.

maxamillion committed:
allow update_cache as stand alone operation for yum/dnf (#46183)

* allow update_cache as stand alone operation for yum/dnf

Fixes #40068

Signed-off-by: Adam Miller <admiller@redhat.com>

* make sanity tests happy

Signed-off-by: Adam Miller <admiller@redhat.com>

evgeni committed:
introduce `module_utils.urls.fetch_file` as a wrapper to download and save files (#19172)

* module_utils.urls: add fetch_file function

* apt: use fetch_file instead of own download()

* unarchive: use fetch_file instead of own codecopy

* apt: add test for deb=http://…

* unarchive: add test for a remote file download and unarchive

* yum: replace fetch_rpm_from_url by fetch_file

* use NamedTemporaryFile

* don't add a dot to fileext, it's already there

Akasurde committed:
Plugin: check if executable exists in local connection (#46201)

This fix checks if executable exists in local connection specified using
ANSIBLE_EXECUTABLE variable.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

colin-mccarthy committed:
Added (Define Values From CSV File) to example section (#46620)

Adding a new example demonstrating setting facts from a CSV file, from a real-world network use case.

s-hertel committed:
Add docs for inventory caching (#46402)

* Add docs for inventory caching

* Fix subsection headers

* Fix top header

samccann committed:
added jinja2 example (#46445)

* added jinja2 example

jborean93 committed:
docs: add Support section for plugin types (#46520)

* docs: add Maintenance section for plugin types
* Added supported_by name in bold to match Status

danieljakots committed:
Register missing parameter reboot_timeout (#46585)

s-hertel committed:
Add 'auto' to documented default enabled inventory plugins (#46621)

* Correct default inventory plugins enabled in docs and example/ansible.cfg

* Fix headers

lwm committed:
Linode maintainers: Add displague. Remove zbal. (#46635)

* Add displague. Remove zbal.

Lines up with https://github.com/ansible/community/issues/365.

RobertHerhold committed:
Update links to module sources (#46553)

The documentation still points to the old links for the [core](https://github.com/ansible/ansible-modules-core/tree/devel) and [extras](https://github.com/ansible/ansible-modules-extras/tree/devel) modules. This PR updates the links to the main ansible repository.

+label: docsite_pr

major committed:
Docs: Fix ftype list in sefcontext module (#46364)

Add the list of file types to the `sefcontext` module documentation.

mattclay committed:
Add link check to `make sdist`.

This will cause `make sdist` to fail on platforms which create
hard links of symbolic links as regular files, such as MacOS (Darwin).

This prevents accidental creation of an sdist tarball without
the necessary symbolic links.

mattclay committed:
Update MANIFEST.in (#46502)

* Update MANIFEST.in:

- Remove unnecessary prune.
- Include files needed by tests.
- Exclude botmeta sanity test.

These changes permit sanity tests to pass on sdist output.

lwm committed:
Add rmcintosh as new Linode maintainer. (#46642)

Part of the push on the new https://github.com/ansible/community/wiki/Linode.

MikeKlebolt committed:
incorrect ref (#46492)

Commit:
feat(apt): add -f --fix-broken option to apt module (#27162)

codebop committed:
merge conflict (#25235)

Benoit-DunandLaisin committed:
Fix python 3 compatibility (#41645)

pilou- committed:
rabbitmq_user: 'node' parameter: add default value (#38156)

* Remove unnecessary workaround

* add test: set RABBITMQ_NODENAME environment variable

The following error occurs:

  TASK [rabbitmq_user : Add user] ***
  fatal: [testhost]: FAILED! => {
    "changed": false,
    "cmd": "/usr/sbin/rabbitmqctl -q list_users",
    "rc": 69,
    "msg": "Error:********@c65c2bc59398'. Please see diagnostics information and suggestions below.\n\nMost common reasons for this are:\n\n * Target node is unreachable (e.g. due to hostname resolution, TCP connection or firewall issues)\n * CLI tool fails to authenticate with the server (e.g. due to CLI tool's Erlang cookie not matching that of the server)\n * Target node is not running\n\nIn addition to the diagnostics info below:\n\n * See the CLI, clustering and networking guides on http://rabbitmq.com/documentation.html to learn more\n * Consult server logs on node test@c65c2bc59398\n\nDIAGNOSTICS\n===========\n\nattempted to contact:********@c65c2bc59398\n * effective user's home directory: /var/lib/rabbitmq\n * Erlang cookie hash: 3MxcYFrJzfhEL+FlUfLlQw==",
    "stderr":  [...],
    "stderr_lines": [
      "Error: unable to perform an operation on node 'test@c65c2bc59398'. Please see diagnostics information and suggestions below.", "",
      "Most common reasons for this are:", "",
      " * Target node is unreachable (e.g. due to hostname resolution, TCP connection or firewall issues)",
      " * CLI tool fails to authenticate with the server (e.g. due to CLI tool's Erlang cookie not matching that of the server)",
      " * Target node is not running", "",
      "In addition to the diagnostics info below:", "",
      " * See the CLI, clustering and networking guides on http://rabbitmq.com/documentation.html to learn more",
      " * Consult server logs on node test@c65c2bc59398", "",
          "DIAGNOSTICS", "===========", "",
          "attempted to contact: [test@c65c2bc59398]", "",
          "test@c65c2bc59398:",
      "  * connected to epmd (port 4369) on c65c2bc59398",
      "  * epmd reports: node 'test' not running at all",
      "                  other nodes on c65c2bc59398: [rabbit]",
      "  * suggestion: start the node", "",
      "Current node details:",
      " * node name: rabbitmqcli2@c65c2bc59398",
      " * effective user's home directory: /var/lib/rabbitmq",
      " * Erlang cookie hash: 3MxcYFrJzfhEL+FlUfLlQw==", ""
    ],
    "stdout": "",
    "stdout_lines": []
  }

* node parameter: fix default value

'rabbit' is the default value mentioned in the module documentation.

jouve committed:
create sources.list.d directory if needed (#36981)

abadger committed:
Mocking out __future__ could cause problems

resmo committed:
botmeta: cloudstack: new team member (#46622)

jborean93 committed:
ansible-test: set ulimit to enforce consistent test environment (#46652)

* ansible-test: set ulimit to enforce consistent test environment

* fixed santiy issue

machacekondra committed:
ovirt_disk: Add wipe_after_delete option (#46679)

lwm committed:
Add a Linode v4 dynamic inventory plugin. (#45902)

* Add a Linode v4 dynamic inventory plugin.

Closes https://github.com/ansible/ansible/issues/44721.

* Use the latest API for accessing host variables.

References:

  * https://github.com/linode/linode_api4-python/issues/141

* Minor docs formating

gundalow committed:
Define team_botmeta & inventory plugin (#46684)

imjoey committed:
ovirt_storage_domain: Fix error message formating (#46678)

mattclay committed:
Use default Shippable container. (#46703)

grimreaper committed:
Reduce noise in warnings from ansible (#40754)

When python is compiled in debug mode, or certain command line flags are
passed, python issues helpful warnings to let users know of files opened
by not closed.

This can be fairly spammy on an ansible run.  This change reduces the
number of such warnings by a factor of 10.

bjakubiak committed:
Fixes #23456 (#35171)

dagwieers committed:
Update wording as requested by review

Also improve wording.

dagwieers committed:
Indentation

rdkls committed:
fix: check execute_lock boolean, not lock string var (#45329)

Jorge-Rodriguez committed:
Rabbitmq: Enable communication to management API over HTTPS (#18437)

* Enable communication to management API over HTTPS.
* Specify version added tags to new parameters
* Set proper parameter type.
* Corrected version_added numbers.
* Extracted commons to ansible utils.
* Fix PEP8 error
* Fix documentation extension syntax.

Fixes #22953

mattclay committed:
Fix vmware_inventory unit tests so they run.

jborean93 committed:
win_uri: use variable for httpbin host (#46734)

biohazd committed:
ec2_asg: adding explanation of replace_all_instances (#46694)

* adding explanation of replace_all_instances

How does "replace_all_instances" work? does it increase the number of instances by 1 in the Load balancer and it scales in a server and waits for it to complete its health check, OR   does it terminate a node a that causes an autoscale event to happen and then adds a new server?
+label: docsite_pr

thpica committed:
Do not try to encode metadata if it is None (#46739)

* Do not try to encode metadata if it is None

* Add changelog fragment

* Fix fragment missing EOF line

Commit:
Added iostype details in network ios_facts module (#46599)

paneu committed:
ios_facts: Fix minor aesthetical glitches (#46577)

* ios_facts: Fix LLDP gathering without neighbors

When LLDP is enabled but no neighbors are present, the following
structure is generated:
...
            "ansible_net_neighbors": {
                "null": [
                    {
                        "host": null,
                        "port": null
                    }
                ]
            },
...
If we are not able to find any relevant LLDP information, bail out early
so cases like shown above are handled more gracefully.

* ios_facts: Remove trailing space in lineprotocol

Some Cisco devices (at least CSR1000V 16.6.4) add a space after the
line protocol. This causes the space to be present in the result of
fact gathering:
            "ansible_net_interfaces": {
                "GigabitEthernet2": {
...
                    "lineprotocol": "up ",

Be more clear about scanning the output to avoid this behaviour.

emonty committed:
Add mnaser to OpenStack team (#46753)

Akasurde committed:
Correct usage of message in ConnectionError (#46566)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

dominikholler committed:
Fix os_router ignores enable_snat: no (#45921)

If enable_snat is False, this should be used to build the
request, because the default value in the OpenStack Networking
API is True.

Fixes the issue #45915.

Akasurde committed:
[do

* Add U() in front of URL
* Add period at end of statement
* Spell check

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

gardouille committed:
Add xz-utils requirement (#46159)

* Document xz-utils requirement for deb parameter of apt module

anasbadaha committed:
Add support for IGMP interfaces on onyx switches (#46218)

Signed-off-by: Anas Badaha <anasb@ufm-host08-004.mtr.labs.mlnx>

akatch committed:
Do not append msg to verbose tasks (#46721)

bcoca committed:
manage levels for 'multiple included lists' (#46359)

* manage levels for 'multiple included lists'

fixes #46343

Akasurde committed:
lambda: fix unboundLocalError in lambda (#46729)

This fix initializes current_version to None so that
module proceeds in check mode.

Fixes: #46654

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Fix netconf module_utils dict changed size issue (#46778)

Fixes #46755

Use list() to copy the keys of attribute dict
while iterating over attribute dict.

bcoca committed:
remove default from delegate_facts to inherit (#45492)

* remove default from delegate_facts to inherit

fixes #45456

* test delegate_facts

* added note about inheritance and defaults

* yamllint

cordones committed:
Update eos_config.py (#45810)

Switches will accept the previously used syntax for /32 ACL entries, but they get changed to host ipaddress in the config file.
This results in the config check failing every time (permit ip 1.1.1.1/32 any log != permit ip host 1.1.1.1 any log), resulting in the ACL getting re-created on every single run of the Playbook.
Making this proposed change to the example will make it work in an idempotent manner.
+label: docsite_pr

maxamillion committed:
yum module handle list optional empty strings properly (#46634)

Fixes #46517

Signed-off-by: Adam Miller <admiller@redhat.com>

fgsoap committed:
Update nso_action.py (#46763)

* Update nso_action.py

The module's name should be nso_action.

* Update nso_action.py

Fixed the input field in the example.

ckotte committed:
VMware: Gather extended datastore facts (#46546)

* Add datacenter_name alias
* Add check mode support
* Add extended datastore information

ckotte committed:
VMware: Add check mode support to module vmware_host_firewall_manager (#46266)

dangoscomb committed:
nmcli: fix vlan connection modification Fixes #42322 (#42415)

* ensure optional items are set to empty strings rather than not presented
fix syntax of vlan modification command

* extended tests for nmcli

markafarrell committed:
Enable HTTP Authentication for url lookup (#43467)

* Add url_username and url_password options for url lookup

* Update url.py

Fix documentation

* Fix options documentation 

Default value for url_password and url_username is None not an empty string

* Add version added to documentation

* Break command over multiple lines to fix linting

* Fix more linting

* Update version_added in url.py to 2.8

bcoca committed:
simpler/shorter option names for user

 - internally we can keep long names, but user should not require extra typing

MatrixCrawler committed:
Convert results with to_native in consul_kv plugin (#46551)

* #42851 convert results with to_native

* added missing ANSIBLE_METADATA

* removed unneeded brackets

* * replaced to_native with to_text to avoid getting bytecode

Commit:
Lenovo finetune modules (#46623)

* Making username password etc non mandatory as we use persistence connection.
Some documentation fine tuning also done.

* To update version which was tested against.

* Adding Bacon switch (NE0152T) to device rules

* Qalthos Review catch

* Updating version

* Changing documentation to add Bacon switch. This is effective since 2.8

zharalim committed:
Clean playbook filters doc by adding whitespaces (#46569)

acozine committed:
adds stub API docs in a single file (#46663)

* adds stub API docs in a single file

maxamillion committed:
yum module properly check for None config_file (#46641)

* yum module properly check for None config_file
* add conf_file test cases to yum integration tests

Signed-off-by: Adam Miller <admiller@redhat.com>

ptux committed:
copy - support recursive copying with remote_src (#43998)

* Allow copy module to work with recursive and remote_src #14131

bcoca committed:
added friendlier aliases for auth options (#46816)

* added friendlier aliases for auth options

s-hertel committed:
add more consistent extension matching for inventory plugins (#46786)

* Add consistent extension matching for inventory plugins that support YAML configuration files

* Document extension matching expectations

mattclay committed:
Fix ansible-test custom docker image traceback.

bcoca committed:
add note to inform awx team on change

remyleone committed:
Use constructable with NetBox dynamic inventory (#45913)

remyleone committed:
Add virtualization devices in the inventory (#45728)

remyleone committed:
Fix scaleway_compute dynamic_ip_required (#46256)

mavit committed:
Locate prtdiag even when absent from /usr/bin (#44113)

* Locate prtdiag even when absent from /usr/bin

On Solaris 8 hosts, this prevents fact collection from aborting with:

    Argument 'args' to run_command must be list or string

* Lint fix.

* Style: pass /usr/platform/.../sbin as optional path to get_bin_path().

maxamillion committed:
fix yum proxy username/password handling (#46291)

Fixes #46249

Signed-off-by: Adam Miller <admiller@redhat.com>

justengland committed:
Update win_scheduled_task.py (#46720)

* Update win_scheduled_task.py to document that the duration of a task trigger can be null, which will cause it to run indefinitely - docs update makes use of suboptions

* Add a fix for the validate-modules schema

Commit:
Provide flexibility when retrieving facts (#46148)

* Provide flexibility when retrieving facts

* Check if keys exist before trying to read

cahlchang committed:
[AWS] Support check mode in aws ec2_instance module (#46774)

* fix checkmode

* Added checking test to existing test playbook

* Added test for check mode

abarbare committed:
fix scaleway user data tests (#46570)

* fix scaleway user data tests

* default values

* pep8

acozine committed:
all azure modules are community maintained (#46839)

acozine committed:
all amazon modules are community maintained (#46840)

samccann committed:
mark Cisco ACI modules as certified (#46841)

samccann committed:
mark Cisco ACI modules certified (#46842)

mattclay committed:
Block module ansible imports outside module_utils.

nitzmahone committed:
remove cross-module imports

* duplicated code instead; use module_utils if code-sharing needs increase

dagwieers committed:
WinRM/PSRP: Add support for unreachable

Currently PSRP connection always fail, even if the system is not
reachable. This PR fixes this.

mattclay committed:
Add OUTPUT_DIR env var to integration tests.

ptux committed:
add sample for follow parameter of copy.py (#45765)

* add sample for follow parameter of copy.py

jborean93 committed:
ansible-test: add skip/windows/... alias to skip tests on specific Windows versions (#46845)

* ansible-test: add skip/windows/... alias to skip tests on specific Windows versions

* show what tests were skipped

* changes to logic to only skip if all Windows targets are set to skip

* codestyle improvements

* change warning message based on review

* check args type before running the Windows path

Alb0t committed:
Add Alibaba to RedHat family list. (#46828)

remyleone committed:
Add a netbox team to botmeta (#46817)

willthames committed:
Ensure that k8s_facts always returns resources key (#46733)

Fix bug returning `items` key if NotFound exception is hit

gustavomcarmo committed:
ldap_attr: typecast values to list (#46818)

Adding a value to an attribute or removing a value from an attribute of a LDAP entry throws the exception TypeError: object of type 'filter' has no len()

ckotte committed:
VMware: Improve module vmware_host_config_manager (#46264)

* Improve module description
* Add check mode support and change message

bcoca committed:
use ansible json encoder for results in callbacks (#46830)

* use ansible json encoder for results in callbacks

Qalthos committed:
Add missing connection param to test (#46831)

bcoca committed:
use to_text instead of str in async_status (#45990)

* use to_text instead of str in async_status

* used to_native instead as per feedback

beeankha committed:
Making minor grammatical changes (#46844)

fixes to dev_guide page

bcoca committed:
now clog matches code

bcoca committed:
better information for user from inventory plugins (#46766)

* better information for user from inventory plugins

 - use foreman as example

samccann committed:
change UCS modules to certified (#46848)

sivel committed:
Do not use mutable defaults in FieldAttribute, instead allow supplying a callable for defaults of mutable types. Fixes #46824 (#46833)

samccann committed:
changed F5 module metadata to certified (#46849)

samccann committed:
Mark NIOS modules as certified (#46973)

* updated NIOS community modules to certified

* moved remaining nios modules to certified

samccann committed:
Change netapp ontap and elementsw modules to certified (#46974)

* netapp ontap and elementsw modules to certified

samccann committed:
update cyberark metadata to certified (#46978)

samccann committed:
update sros metadata to certified (#46981)

jborean93 committed:
win test: add http tester container to Windows tests (#46606)

dagwieers committed:
Move requests import into pypsrp try-block

dagwieers committed:
Add changelog snippet

dagwieers committed:
Update changelog fragment

dagwieers committed:
Add ConnectionError too

mattclay committed:
Move unit test compat code out of `lib/ansible/`. (#46996)

* Move ansible.compat.tests to test/units/compat/.
* Fix unit test references to ansible.compat.tests.
* Move builtins compat to separate file.
* Fix classification of test/units/compat/ dir.

IAmHughes committed:
Fix spelling issue (#46982)

Corrects spelling from `fo` to `of` for `storage_blob_name`

+label: docsite_pr

bingoarun committed:
Added ability to create Openstack users without password (#37338)

bingoarun committed:
Fixes #35284 - Added ability for os_nova_flavor to update existing flavors (#37170)

ckotte committed:
VMware: Improve module vmware_host_datastore (#45652)

* Don't execute mount or umount in check mode
* The datastore was mounted or unmounted when you execute the playbook in
check mode.
* Test if running in check mode before mounting or unmounting the
datastore.
* Add support for NFS v4.1 datastores

mnaser commented on lib/ansible/modules/cloud/openstack/os_server.py:
```diff
@@ -132,6 +132,13 @@
+         Equivalent of files= option in nova boot for personality
+     required: false
+     default: null
+     version_added: "2.4"
```
Can you make this 2.8 please?

mnaser reviewed (changes requested):
Sorry for the delay in reviewing this, I'll be able to merge on this is taken care of.

samccann committed:
WIP: fix build errors due to missing certified_supported label (#46835)

* fix build errors due to missing certified_supported label

* fixed references and toc

jctanner committed:
Add a test target for the foreman inventory script (#46837)

Add a test target for the foreman inventory script

justjais committed:
Fix to resolve bug 46671, which now handles checking for Nonetype connection prompt (#46677)

* added Nonetype check

* implemented review comment

* review changes

mattclay committed:
Upgrade pylint to version 2.1.1. (#47036)

* Fix issues reported by the latest pylint.
* Split pylint runs into more contexts.
* Upgrade pylint.

ganeshrn committed:
Fix in confirmed_commit capability in netconf_config modules (#46964)

* Fix in confirmed_commit capability in netconf_config modules

Fixes #46804

*  If confirm value is greater than zero or confirm_commit option is set and confirmed-commit
   capability is not supported but Netconf server only in that case fail the module

* Update confirm-commit flag

* Minor changes

mattclay committed:
Add constraint for pycparser.

mattclay committed:
Add sphinx constraint for rstcheck compatibility.

Akasurde committed:
VMware: Provide facility to specify path for template (#38031)

When multiple virtual machines or templates are found in environment,
module needs folder path. This fix adds facility to specify folder path
for template or virtual machine if multiples are found.

Fixes: #37087

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

healem committed:
VMware: Added secure boot enable/disable to vmware_guest_boot_manager. (#46717)

* Added secure boot enable/disable to vmware_guest_boot_manager.  Also added its state to facts
* VMware does not support secure boot when boot_firmware is bios.  Add some guardrails to protect the user from misconfiguration
* Address review comments

justjais committed:
revert 46677 changes (#47044)

acozine committed:
fixes local toc, updates language (#47054)

ptux committed:
option [others] of get_url is unnecessary (#47005)

* parameters [others] of get_url is unnecessary

* delete a blank line

* test

* delete ignore

* Update get_url.py

mattclay committed:
Use 4 nodes for sanity tests in CI.

mattclay committed:
Use 4 nodes for sanity tests in CI.

Commit:
Adding mysql firewall rule module (#45074)

* adding mysql firewall rule

* fixed mysql firewall rule

* use mgmtclient from common

* fix pep8

* small updates

* minor docs fix

Commit:
adding postgresql firewall rule (#45077)

* adding postgresql firewall rule

* fixed test problems

* add state into the doc

* use lropoller

* updated samples and some naming

sdodsley committed:
Add new Pure Storage maintainers (#47074)

mattclay committed:
Add BOM check to shebang sanity test.

Includes fixes for docsite files with a UTF-8 BOM.

Jorge-Rodriguez committed:
rabbitmq_binding: Fix misnamed reference (#47049)

Fix for bug introduced in #18437

Akasurde committed:
efs_facts: Documentation update (#46826)

* Update pydoc
* Update module documentation and example
* Add aliases for name as creation_token

Fixes: #28738

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Revert "Use 4 nodes for sanity tests in CI."

This reverts commit b887333029e6a6f1e3e8d7577711297812e97ebc.

mattclay committed:
Fix ansible-test --exclude with delegation.

Previously the option worked with integration commands but not units or sanity.

sivel committed:
Use the copied and merged task for calculating task vars in the free strategy. Fixes #47024 (#47060)

jborean93 committed:
ansible-test: run win httptester with bypass policy (#47090)

Commit:
bigip_firewall_rule -- missing port_range (#47077)

port_range was missing from bigip_firewall_rule

jctanner committed:
Handle non-ascii characters in foreman.py (#46779)

* Handle non-ascii characters in foreman.py

* Add test to validate non-ascii results

mjhoffman65 committed:
adds redirect configurations and probes to azure_rm_appgateway (#46607)

jborean93 committed:
ansible-test: setup up http runner in between each target (#47100)

* ansible-test: setup up http runner in between each target

* review changes

imjoey committed:
ovirt_storage_domain: Improve logic for state controlling. (#46951)

fxfitz committed:
Add unit tests for junos terminal (#47103)

sdodsley committed:
BOTMETA : Add more Pure maintainers (#47116)

Akasurde committed:
VMware: new module: vmware_guest_disk (#36165)

1. Add support for SCSI controller creation
2. Support for SCSI Type
3. Warn about Disk Unit Number 7
4. Allow specifying SCSI controller
5. Allow specifying Disk Unit Number
6. Idempotency
7. Remove disks
8. Do not allow to reduce disk
9. More than 15 disks support
10. No SCSI controller and unit number check
11. Thin support
12. Update documentation and example
13. Multiple Datastore and datastore cluster support
14. Check datatype of disk unit number and SCSI controller
15. Handle disk_size when it is int or float

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

pabowers committed:
Update the document to link to the script itself (#46986)

Added a link to the dynamic inventory script to make it easier for a user to find.

sivel committed:
Add porting guide note about inline vars on import/include_tasks (#47121)

* Add porting guide note about inline vars. Fixes #47102

mattclay committed:
Update default-test-container to version 1.4.1. (#47042)

mmazur committed:
Correct wording in k8s module description (#47125)

dagwieers committed:
This fixes a weird indentation issue in the menu (#45089)

sivel committed:
Ensure that an empty literal list with loop skips the task (#47129)

jborean93 committed:
test: openssl 1.1.x compatibility (#47112)

jvandermeulen committed:
Fixed typo (#47139)

<!--- fixed typo in exclude example -->

+label: docsite_pr

mattclay committed:
Update requirements for urllib3 for python 2.6.

ruzickap committed:
with_items is obsolete - loop should be used instead (#46524)

ruzickap committed:
with_items is obsolete - using loop instead (#46805)

ruzickap committed:
with_items is obsolete - using loop instead (#46806)

userlerueda committed:
Update os_server.py

userlerueda committed:
Update os_server.py

userlerueda committed:
Update lib/ansible/modules/cloud/openstack/os_server.py

Integrate fix for transferring files

userlerueda committed:
Merge branch 'issue-22244' of github.com:userlerueda/ansible into issue-22244

Fix for conflicts

userlerueda commented:
Closing this PR and opening another one (#47196). devel is too far away in changes from the proposed change.
